src/libraries/System.Runtime/tests/TrimmingTests/AppDomainGetThreadPrincipalTest.cs

```diff
@@ -0,0 +1,14 @@
+using System.Security.Principal;
+using System.Threading;
+
+class Program
+{
+    static int Main(string[] args)
+    {
+        // Thread.CurrentPrincipal calls AppDomain.CurrentDomain.GetThreadPrincipal() which
```

Comment:
This path is taken only when you set PrincipalPolicy to `PrincipalPolicy.WindowsPrincipal` or `PrincipalPolicy.UnauthenticatedPrincipal`. As written, this test won't exercise these codepaths.